### PR TITLE
Added session force disconnect feature

### PR DESCRIPTION
**Pull request by dergraf**
Todo: 
- Add integration with vernemq_dev

### CODE REVIEW

**Comment by dergraf** [comment]
@larshesel what do you think of this approach?

**Comment by larshesel** [comment]
Looks pretty good to me - I noticed that at least one of the new tests failed on travis ...

File: apps/vmq_server/test/vmq_queue_SUITE.erl [hunk 1]
```diff
@@ -236,6 +240,58 @@ queue_persistent_client_expiration_test(_) ->
     not_found = vmq_queue_sup_sup:get_queue_pid(SubscriberId),
     {ok, []} = vmq_lvldb_store:msg_store_find(SubscriberId).
 
+queue_force_disconnect_test(_) ->
+    Parent = self(),
+    SubscriberId = {"", <<"force-client-disconnect">>},
+    QueueOpts = maps:merge(vmq_queue:default_opts(), #{clean_session => false,
+                                                       max_offline_messages => 1000,
```
Comment (codeadict) [inline apps/vmq_server/test/vmq_queue_SUITE.erl]: Should this be a hardcoded 1000 or use the User's setting?

Comment (dergraf) [reply apps/vmq_server/test/vmq_queue_SUITE.erl]: you're right, for this test case it really doesn't matter.

File: apps/vmq_server/src/vmq_info_cli.erl [hunk 2]
```diff
@@ -92,6 +128,7 @@ session_usage() ->
      "  Retrieve information on live sessions.\n\n",
```
Comment (larshesel) [inline apps/vmq_server/src/vmq_info_cli.erl]: This should probably be changed to something else, like `Manage MQTT sessions`.

File: apps/vmq_server/src/vmq_info_cli.erl [hunk 3]
```diff
@@ -92,6 +128,7 @@ session_usage() ->
...
      "  Sub-commands:\n",
      "    show        Show and filter running sessions\n",
+     "    disconnect  Forcefully disconnect a session\n",
```
[the remark this answers is not in the document]
Comment (drf) [reply apps/vmq_server/src/vmq_info_cli.erl]: "cleanup  Forcefully clean up a session (will disconnect the client if connected)"

File: apps/vmq_server/src/vmq_queue.erl [hunk 4]
```diff
@@ -520,6 +527,25 @@ handle_sync_event(get_sessions, _From, StateName, #state{sessions=Sessions} = St
     {reply, maps:keys(Sessions), StateName, State};
 handle_sync_event(get_opts, _From, StateName, #state{opts=Opts} = State) ->
     {reply, Opts, StateName, State};
+handle_sync_event({force_disconnect, DoCleanup}, _From, StateName,
+                  #state{id=SId, sessions=Sessions, offline=#queue{queue=OfflineQ}} = State) ->
+    %% Forces to disconnect all sessions and to cleanup all state
+    %% session has expired cleanup and go down
```
Comment (larshesel) [inline apps/vmq_server/src/vmq_queue.erl]: The above line looks like an old cut'n'paste leftover. Should be removed.

File: apps/vmq_server/src/vmq_queue.erl [hunk 5]
```diff
@@ -520,6 +527,25 @@ handle_sync_event(get_sessions, _From, StateName, #state{sessions=Sessions} = St
     {reply, maps:keys(Sessions), StateName, State};
 handle_sync_event(get_opts, _From, StateName, #state{opts=Opts} = State) ->
     {reply, Opts, StateName, State};
+handle_sync_event({force_disconnect, DoCleanup}, _From, StateName,
+                  #state{id=SId, sessions=Sessions, offline=#queue{queue=OfflineQ}} = State) ->
+    %% Forces to disconnect all sessions and to cleanup all state
```
Comment (larshesel) [inline apps/vmq_server/src/vmq_queue.erl]: I think this would read nicer like this: `Forcefully disconnect all sessions and cleanup all state`

**Pull request merged**